ios/Exponent/Versioned/Core/Api/EXNotifications.m

```diff
@@ -92,41 +93,116 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId kernelServiceDeleg
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(__unused RCTPromiseRejectBlock)reject)
 {
-  UILocalNotification *notification = [self _localNotificationFromPayload:payload];
 
+  UNMutableNotificationContent* content = [self _localNotificationFromPayload:payload];
   [EXUtil performSynchronouslyOnMainThread:^{
-    [RCTSharedApplication() presentLocalNotificationNow:notification];
+    UNTimeIntervalNotificationTrigger* trigger = [UNTimeIntervalNotificationTrigger triggerWithTimeInterval:1 repeats:NO];
+    UNNotificationRequest* request = [UNNotificationRequest
+                                      requestWithIdentifier:content.userInfo[@"id"] content:content trigger:trigger];
+    [[UNUserNotificationCenter currentNotificationCenter] addNotificationRequest:request withCompletionHandler:^(NSError * _Nullable error) {
+      if (error != nil) {
+        NSLog(@"%@", error.localizedDescription);
+        reject(@"Could not make notification request", error.localizedDescription, error);
+      } else {
+        resolve(content.userInfo[@"id"]);
+      }
+    }];
   }];
 
-  resolve(notification.userInfo[@"id"]);
+}
+
+RCT_EXPORT_METHOD(addCategory: (NSString *) categoryId
+                  actions: (NSArray *) actions
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(__unused RCTPromiseRejectBlock)reject)
+{
+  NSMutableArray<UNNotificationAction *> * actionsArray = [[NSMutableArray alloc] init];
+  
+  for( NSArray * action in actions) {
+    int optionsInt = [(NSNumber *)action[2] intValue];
+    UNNotificationActionOptions options = UNNotificationActionOptionForeground + optionsInt;
+                                     
+    if (([action count] == 5)) {
+      UNTextInputNotificationAction * newAction = [UNTextInputNotificationAction actionWithIdentifier:action[0]
+                                                                                                title:action[1]
+                                                                                              options:options
+                                                                                 textInputButtonTitle:action[3] textInputPlaceholder:action[4]];
+      [actionsArray addObject:newAction];
+    } else {
+      UNNotificationAction * newAction = [UNNotificationAction actionWithIdentifier:action[0] title:action[1] options:options];
+      [actionsArray addObject:newAction];
+    }
+  }
+  
+  UNNotificationCategory * newCategory = [UNNotificationCategory categoryWithIdentifier:categoryId actions:actionsArray intentIdentifiers:@[] options:UNNotificationCategoryOptionNone];
+  
+  [[UNUserNotificationCenter currentNotificationCenter] setNotificationCategories:[NSSet setWithObject:newCategory]];
+  resolve(@"done");
```

Comment:
`@"done"` isn't the best return value, `actionId` would work better or just `@(YES)`